src/shared/services/katex.service.ts

```diff
@@ -0,0 +1,168 @@
+import { Injectable } from '@angular/core';
+import { List, Map } from 'immutable';
+import { render, ParseError } from 'katex';
+
+import { JSONSchema } from '../interfaces';
+
+@Injectable()
+export class KatexService {
+  private delimiters = [
+    { left: '$$', right: '$$', display: true },
+    { left: '$', right: '$', display: false },
+    // FIXME check if we need the ones below
+    { left: '\\[', right: '\\]', display: true },
+    { left: '\\(', right: '\\)', display: false },
+  ];
+
+  renderMathInText(text: string, el: HTMLElement) {
```

Comment:
this one needs some documentation, so that we know it renders latex text into a given html element.